lib/iris/experimental/ugrid/mesh.py

```diff
@@ -422,42 +424,42 @@ def indices_error(message):
 
         len_req_fail = False
         if shapes_only:
-            src_shape = indices_shape[self.tgt_dim]
+            location_shape = indices_shape[self.connected_axis]
             # Wrap as lazy to allow use of the same operations below
             # regardless of shapes_only.
-            src_lengths = _lazy.as_lazy_data(np.asarray(src_shape))
+            location_lengths = _lazy.as_lazy_data(np.asarray(location_shape))
         else:
             # Wouldn't be safe to use during __init__ validation, since
-            # lazy_src_lengths requires self.indices to exist. Safe here since
+            # lazy_location_lengths requires self.indices to exist. Safe here since
             # shapes_only==False is only called manually, i.e. after
             # initialisation.
-            src_lengths = self.lazy_src_lengths()
-        if self.src_location in ("edge", "boundary"):
-            if (src_lengths != 2).any().compute():
+            location_lengths = self.lazy_location_lengths()
+        if self.location_element in ("edge", "boundary"):
+            if (location_lengths != 2).any().compute():
                 len_req_fail = "len=2"
         else:
-            if self.src_location == "face":
+            if self.location_element == "face":
                 min_size = 3
-            elif self.src_location == "volume":
-                if self.tgt_location == "edge":
+            elif self.location_element == "volume":
+                if self.connected_element == "edge":
                     min_size = 6
                 else:
                     min_size = 4
             else:
                 raise NotImplementedError
-            if (src_lengths < min_size).any().compute():
+            if (location_lengths < min_size).any().compute():
                 len_req_fail = f"len>={min_size}"
         if len_req_fail:
             indices_error(
-                f"Not all src_locations meet requirement: {len_req_fail} - "
+                f"Not all location_elements meet requirement: {len_req_fail} - "
                 f"needed to describe '{self.cf_role}' ."
             )
 
     def validate_indices(self):
         """
         Perform a thorough validity check of this connectivity's
         :attr:`indices`. Includes checking the sizes of individual
-        :attr:`src_location`\\ s (specified using masks on the
+        :attr:`location_element`\\ s (specified using masks on the
```

Comment:
as above, replace with 'location elements'. Also, consider something more like "checking the sizes of connectivities associated with individual 'location elements'"

Reply:
b01341dce90fb5f1e2c7d33aa5d44152af2924a7